### DISCUSSION TITLE
[Question]: How can I get Mongo on ARM v7?

### DISCUSSION THREAD

**Discussion by Jaxx7594** [danny-avila/LibreChat#1810] [Troubleshooting] [answered]
### What is your question?

I've been trying to set up Libre on my Raspberry Pi 4B 8GB, and I keep encountering this error when running docker-compose up: 
```
ERROR: no matching manifest for linux/arm/v7 in the manifest list entries
```
How can I fix this? I can't find any information online about it.

### More Details

I tried using version 4.4.18 as shown in the documentation, but the error still occurs. I then attempted to use older versions before [MongoDB deprecated 32 bit builds](https://jira.mongodb.org/browse/DOCS-5815), but I get this error:
````
npm WARN deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
npm WARN deprecated domexception@4.0.0: Use your platform's native DOMException instead
npm notice
npm notice New minor version of npm available! 10.2.3 -> 10.4.0
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v10.4.0>
npm notice Run `npm install -g npm@10.4.0` to update!
npm notice
npm ERR! code 1
npm ERR! path /app/node_modules/sharp
npm ERR! command failed
npm ERR! command sh -c (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
npm ERR! sharp: Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.14.5/libvips-8.14.5-linuxmusl-armv7.tar.br
npm ERR! sharp: Please see https://sharp.pixelplumbing.com/install for required dependencies
npm ERR! sharp: Installation error: Prebuilt libvips 8.14.5 binaries are not yet available for linuxmusl-armv7

npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2024-02-15T05_52_49_018Z-debug-0.log
````

### What is the main subject of your question?

Installation

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment]
not sure if this will work, but you can try to specify the platform in the override file

example:
```yaml
services:
  service_name:
    image: example/image
    platform: linux/arm64
```

**Comment by Jaxx7594** [comment]
Apologies for the delay, my raspberry pi has lots of other things running on it so it took a while to ssh and make those changes. Got this error:
```
ERROR: The Compose file './docker-compose.override.yml' is invalid because:
Unsupported config option for services.mongodb: 'platform'
```

**Comment by fuegovic** [comment]
try specifying the platform for librechat, and not specifying it for mongo

**Comment by Jaxx7594** [comment]
Could you give me an example of what you mean? I'm not very familiar with docker and its yml syntax

**Comment by fuegovic** [comment]
I'd try something like that. We recently added dedicated arm docker image, that might be why you have an issue. But I don't have a rpi (or a arm device) to test, so I'm not sure either 
```yaml
services:
  api:
    volumes:
      - ./librechat.yaml:/app/librechat.yaml
    image: ghcr.io/danny-avila/librechat-dev:latest
    platform: linux/arm64
  mongodb:
    image: mongo:4.4.18
```

**Reply by skyhi69** [reply]
May I ask where is this image? I can not find it.

**Comment by Jaxx7594** [comment]
Still getting an error:
```
ERROR: The Compose file './docker-compose.override.yml' is invalid because:
Unsupported config option for services.api: 'platform'
```

**Reply by fuegovic** [reply]
try validating your file: https://yamlchecker.com/

**Reply by Jaxx7594** [reply]
Apologies for the delay, haven't had time to troubleshoot until now.
![image](https://github.com/danny-avila/LibreChat/assets/101913901/094428ae-56fb-4ad2-b0cc-e87f66a33f2f)

**Comment by Jaxx7594** [comment, marked as answer]
Looks like the root cause is that I'm running raspberry pi OS 32bit instead of 64bit, and mongo only supports 64bit. 32bit is the default, and I guess I didn't realise before I set everything up. Unless anyone has a better plan, I'm gonna get started on updating to 64bit and reinstalling everything... *sigh*

**Comment by Jaxx7594** [comment]
Finally done reinstalling everything... and its still erroring. Not sure what this error means.
```
=> ERROR [api 5/6] RUN apk --no-cache add curl &&     npm ci                                                     10.8s
------
 > [api 5/6] RUN apk --no-cache add curl &&     npm ci:
0.415 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/main/aarch64/APKINDEX.tar.gz
5.423 WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.19/main: temporary error (try again later)
5.423 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/community/aarch64/APKINDEX.tar.gz
10.43 WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.19/community: temporary error (try again later)
10.43 ERROR: unable to select packages:
10.43   curl (no such package):
10.43     required by: world[curl]
------
failed to solve: process "/bin/sh -c apk --no-cache add curl &&     npm ci" did not complete successfully: exit code: 1
```

**Comment by Jaxx7594** [comment]
fixed it with [this](https://stackoverflow.com/questions/74288981/docker-build-error-https-dl-cdn-alpinelinux-org-alpine-v3-14-community-tempo), but now I can't get past the login page using social signup. clicking on the button works fine, but it redirects to localhost, even though I'm accessing it via ipv4 on a different computer. DOMAIN_CLIENT and DOMAIN_SERVER are both set to 192.168.20.23:3080. Is that the correct way to set it up?

**Reply by Jaxx7594** [reply]
Nevermind! I forgot to change the callback url. My bad.

**Comment by Jaxx7594** [comment]
Everything is working now, I'll close this discussion

**Reply by fuegovic** [reply]
Happy to hear you got it working!

**Comment by deepanchal** [comment]
In my case, I added this to my `docker-compose.override.yml` and it worked.
Reference: https://github.com/linuxserver/docker-unifi-network-application/issues/4#issuecomment-1719400553

```yaml
version: '3.4'

services:
  mongodb:
    # 4.4.18 is the last version that runs on RPi's ARM.
    image: arm64v8/mongo:4.4.18
```

**Reply by skyhi69** [reply]
Sorry for my ignorance, I am new and trying to figure out how to install librechat on my rpi 4.

I searched the github page and can not find a docker-compose.override.yml for the proper image. Does anyone have a complete docker compose image for a rpi 4 I can copy?

**Reply by deepanchal** [reply]
@skyhi69 Here's my librechat setup for my rpi4 you can use:
https://github.com/deepanchal/self-hosted/blob/main/apps/librechat
